CelesteNet.Client/Components/CelesteNetMainComponent.cs

```diff
@@ -908,6 +881,22 @@ public CelesteNetMainComponent(CelesteNetClientContext context, Game game)
 
         #endregion
 
+        public void ResetState(Player player = null, Session ses = null) {
+            // Clear ghosts if the scene changed
+            if (player?.Scene != Player?.Scene) {
```

Comment:
Oh I can add these types of comments too, eh. 

Forgot to mention this line probably needs to be changed to something like:
```
if (player != null && player.Scene != Player?.Scene) {
```
just so we don't delete all Ghosts just because Player is null, probably happens somewhere during respawning.

Reply:
Actually my suggestion would change/break Cleanup() behavior, which sets Player to null and removes all Ghosts -- with my suggestion, the Ghost removal wouldn't happen anymore...
Maybe we should seperate the Ghost removal from ResetState?